### DISCUSSION TITLE
How to remove objects with a delete marker?

### DISCUSSION THREAD

**Discussion by artworkad** [minio/minio#15905] [Q&A] [answered]
Hello,

I am new to minio. I wanted to do site replication for backups. Site-replications automatically enables versioning. Now, when an object is deleted, it gets a delete marker as its recent version. 

Is there anything I can configure in minio (UI), to completly remove deleted files after say 30 days?

**Comment by klauspost** [comment, marked as answer]
Set [--noncurrentversion-expiration-days](
https://min.io/docs/minio/linux/reference/minio-mc/mc-ilm-add.html#mc.ilm.add.-noncurrentversion-expiration-days) to 30 and set [--expired-object-delete-marker](https://min.io/docs/minio/linux/reference/minio-mc/mc-ilm-add.html#mc.ilm.add.-expired-object-delete-marker)

**Reply by garry-t** [reply]
what about terraform rule? can't figure out how to set such rule
